sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1119,7 +1119,8 @@ class DataFrame private[sql](
   def explode[A <: Product : TypeTag](input: Column*)(f: Row => TraversableOnce[A]): DataFrame = {
     val schema = ScalaReflection.schemaFor[A].dataType.asInstanceOf[StructType]
 
-    val elementTypes = schema.toAttributes.map { attr => (attr.dataType, attr.nullable) }
+    val elementTypes = schema.toAttributes.map {
+      attr => (attr.dataType, attr.nullable, Some(attr.name)) }
```

Comment:
As we specify the names here, we don't need to pass names to `Generate` as `UnresolvedAttribute` at the end of this method anymore.